### PR TITLE
added config saving

### PR DESCRIPTION
**Pull request by NopoTheGamer**

### CODE REVIEW

File: src/main/java/io/github/moulberry/notenoughupdates/commands/repo/ReloadRepoCommand.java [hunk 1]
```diff
@@ -24,17 +24,17 @@ public void processCommand(ICommandSender sender, String[] args) throws CommandE
...
-				BufferedReader reader = new BufferedReader(new InputStreamReader(
-					new FileInputStream(NotEnoughUpdates.INSTANCE.getConfigFile()),
-					StandardCharsets.UTF_8
-				))
-			) {
-				NotEnoughUpdates.INSTANCE.config = NotEnoughUpdates.INSTANCE.manager.gson.fromJson(reader, NEUConfig.class);
-			} catch (Exception ignored) {
-			}
-		}
+//		NotEnoughUpdates.INSTANCE.newConfigFile();
```
Comment (romangraef) [inline src/main/java/io/github/moulberry/notenoughupdates/commands/repo/ReloadRepoCommand.java:27]: why just comment it out and not straight up remove it

File: src/main/java/io/github/moulberry/notenoughupdates/miscgui/GuiEnchantColour.java [hunk 2]
```diff
@@ -422,6 +422,9 @@ private void renderSideBar(int mouseX, int mouseY, float partialTicks) {
 
 	@Override
 	protected void keyTyped(char typedChar, int keyCode) throws IOException {
+		if (Keyboard.isKeyDown(Keyboard.KEY_ESCAPE)) {
+			NotEnoughUpdates.INSTANCE.saveConfig();
```
Comment (romangraef) [inline src/main/java/io/github/moulberry/notenoughupdates/miscgui/GuiEnchantColour.java:426]: why not put this in `onClose()V`

File: src/main/java/io/github/moulberry/notenoughupdates/miscgui/GuiInvButtonEditor.java [hunk 3]
```diff
@@ -822,6 +823,9 @@ protected void mouseClicked(int mouseX, int mouseY, int mouseButton) throws IOEx
 
 	@Override
 	protected void keyTyped(char typedChar, int keyCode) throws IOException {
+		if (Keyboard.isKeyDown(Keyboard.KEY_ESCAPE)) {
+			NotEnoughUpdates.INSTANCE.saveConfig();
```
Comment (romangraef) [inline src/main/java/io/github/moulberry/notenoughupdates/miscgui/GuiInvButtonEditor.java:827]: why not put this in `onClose()V`

File: src/main/java/io/github/moulberry/notenoughupdates/miscgui/GuiEnchantColour.java [hunk 4]
```diff
@@ -422,6 +422,9 @@ private void renderSideBar(int mouseX, int mouseY, float partialTicks) {
 
 	@Override
 	protected void keyTyped(char typedChar, int keyCode) throws IOException {
+		if (Keyboard.isKeyDown(Keyboard.KEY_ESCAPE)) {
```
Comment (Lulonaut) [inline src/main/java/io/github/moulberry/notenoughupdates/miscgui/GuiEnchantColour.java:425]: Using onGuiClose would probably be better

**Review by romangraef — APPROVED** [review]

**Review by Lulonaut — CHANGES_REQUESTED** [review]

File: src/main/java/io/github/moulberry/notenoughupdates/miscgui/GuiEnchantColour.java (hunk 4 above)
Comment (romangraef) [reply src/main/java/io/github/moulberry/notenoughupdates/miscgui/GuiEnchantColour.java:425]: Ha, first

**Comment by Lulonaut** [comment]
oops to slow

**Review by Lulonaut — APPROVED** [review]

**Review by romangraef — APPROVED** [review]

**Review by DeDiamondPro — APPROVED** [review]

**Pull request merged**